src/mock/browser/containers.ts

```diff
@@ -0,0 +1,56 @@
+import type {ContextualIdentities as CI} from 'webextension-polyfill';
+
+import * as events from '../events';
+
+class MockContainers implements CI.Static {
```

Comment:
Are you planning to write tests? (I assume so since you added the mock but wanted to double-check.)

I'm trying to think how to integrate this with the window/tab mocks, and I don't thiiiink it's too hard? The main dependency is the tab's `cookieStoreId` it seems like. So you might even be able to get away with keeping the two completely independent (though it wouldn't have the most realistic behavior--what happens if a container is deleted while it has an open tab, for example?).

Reply:
> Are you planning to write tests? (I assume so since you added the mock but wanted to double-check.)

Wellll, I can if you'd like me to. If you let me get away with it, I'd probably take the lazy approach. :)

I actually wrote those mocks when the existing tests failed, something to do with events (it's been a while) because I thought the problem was the mocks for not existing which turned out to be wrong.

Reply:
> Wellll, I can if you'd like me to. If you let me get away with it, I'd probably take the lazy approach. :)

LOL, fair. I think we can probably get away without tests for now (which means we don't need the mock either). As much as I enjoy the "make coverage number go up" game, I think the place where tests really help is where you're maintaining some kind of non-trivial invariant, and that's not really the case here like it is in some of the other models (tabs and bookmarks).

> I actually wrote those mocks when the existing tests failed, something to do with events (it's been a while) because I thought the problem was the mocks for not existing which turned out to be wrong.

Huh, I try to make sure `master`'s tests are rock-solid, so if you DO see an unrelated test failure, please do open a bug. :)

> By the way, sorry for all the other minor style stuff that snuck through. I thought I did a self-review on this one but clearly I missed stuff that could have saved you some time!

No worries! Sometimes it happens no matter how careful any of us are, and that stuff is the easiest to fix. :)

Reply:
>  I think we can probably get away without tests for now (which means we don't need the mock either).

Sweet. You wouldn't want to leave the mock infrastructure in place so that if things change in the future it would be at least possible to start making testings without having to repeat that work? If you do want to just remove it, I'll save it somewhere so it'll be theoretically available as a starting point for container tests.

> Huh, I try to make sure master's tests are rock-solid, so if you DO see an unrelated test failure

I'm 98% sure it wasn't your fault, don't worry! Unfortunately, I just can't remember exactly what the problem was. Something about the GC function.

> No worries! Sometimes it happens no matter how careful any of us are,

I can't claim to have been super careful, but thanks. :)

Reply:
I guess it doesn't hurt anything to leave it, especially since it's in its own file.